packages/app/src/components/catalog/EntityPage.tsx

```diff
@@ -260,14 +263,53 @@ const DefaultEntityPage = ({ entity }: { entity: Entity }) => (
   </EntityPageLayout>
 );
 
+const ApiOverviewContent = ({ entity }: { entity: Entity }) => (
+  <Grid container spacing={3}>
+    <Grid item md={6}>
+      <AboutCard entity={entity} />
+    </Grid>
+  </Grid>
+);
+
+const ApiDefinitionContent = ({ entity }: { entity: Entity }) => (
+  <Grid container spacing={3}>
+    <Grid item xs={12}>
+      <ApiDefinitionCard apiEntity={entity} />
+    </Grid>
+  </Grid>
+);
+
+const ApiEntityPage = ({ entity }: { entity: Entity }) => (
+  <EntityPageLayout>
+    <EntityPageLayout.Content
+      path="/*"
+      title="Overview"
+      element={<ApiOverviewContent entity={entity} />}
+    />
+    <EntityPageLayout.Content
+      path="/definition/*"
+      title="Definition"
+      element={<ApiDefinitionContent entity={entity} />}
+    />
+  </EntityPageLayout>
+);
+
 export const EntityPage = () => {
-  const { entity } = useEntity();
-  switch (entity?.spec?.type) {
-    case 'service':
-      return <ServiceEntityPage entity={entity} />;
-    case 'website':
-      return <WebsiteEntityPage entity={entity} />;
-    default:
-      return <DefaultEntityPage entity={entity} />;
-  }
+  return (
+    <EntityPageSwitch>
+      <EntityPageSwitch.Variant
+        kind="Component"
+        type="service"
+        page={ServiceEntityPage}
+      />
+      <EntityPageSwitch.Variant
+        kind="Component"
+        type="website"
+        page={WebsiteEntityPage}
+      />
+      <EntityPageSwitch.Variant kind="Component" page={DefaultEntityPage} />
+      <EntityPageSwitch.Variant kind="API" page={ApiEntityPage} />
+      <EntityPageSwitch.Variant page={DefaultEntityPage} />
+    </EntityPageSwitch>
+  );
```

Comment:
@freben I did some more experiments. What do you think about something like this? Not sure about my implementation though 😉

Reply:
Update on composability experiments! https://github.com/backstage/backstage/issues/1536#issuecomment-729621634 :>

We'll most likely bring all different entity types in under a single route, and then use filters/switches to select what entity pages end up being rendered. It's pretty much exactly what you got here though :grin:

https://github.com/backstage/backstage/blob/2488367b58740602e813de2ef28e3552ca91644f/plugins/explore/src/components/Experiment11.tsx#L224

Reply:
@Rugvip  Cool, give me an update when I should continue with this PR. Thanks!

Reply:
I have to admit that it sounds a bit like this experiments are a little bit more down the road than I would hope. Should I remove the last commit that introduced the `EntityPageSwitch` from this PR and continue with it in the old style? Sure it's more code for now, but it's not a problem that I should solve in this PR. Than I would polish the rest of the PR and we can continue with it.

@Rugvip What do you think?

Reply:
Yeah think it makes sense to go with the old approach for now and then have another look later